src/kernel/include/thd.h

```diff
@@ -13,68 +13,16 @@
 #include "fpu_regs.h"
 #include "chal/cpuid.h"
 #include "chal/call_convention.h"
+#include "chal/chal_thd.h"
```

Comment:
Removing the `struct thd` is a little extreme here. I would have assumed that removing only parts of the thread that are arch-specific would be possible.